flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorGateway.java

```diff
@@ -175,4 +177,13 @@
 		final AllocationID allocationId,
 		final Throwable cause,
 		@RpcTimeout final Time timeout);
+
+	/**
+	 * Requests the file upload of the specified type to the cluster's {@link BlobServer}.
+	 *
+	 * @param fileType to upload
+	 * @param timeout for the asynchronous operation
+	 * @return Future which is completed with the {@link TransientBlobKey} of the uploaded file.
+	 */
+	CompletableFuture<TransientBlobKey> requestFileUpload(FileType fileType, @RpcTimeout Time timeout);
```

Comment:
Don't we need to clean up these methods?
```
	@Override
	public CompletableFuture<TransientBlobKey> requestTaskManagerLog(Time timeout) {
//		return taskExecutorGateway.requestTaskManagerLog(timeout);
		throw new UnsupportedOperationException("Operation is not yet supported.");
	}

	@Override
	public CompletableFuture<TransientBlobKey> requestTaskManagerStdout(Time timeout) {
//		return taskExecutorGateway.requestTaskManagerStdout(timeout);
		throw new UnsupportedOperationException("Operation is not yet supported.");
	}
```

Reply:
Not strictly, but I thought that we could combine both methods in `requestFileUpload` which will give us a leaner interface.

Reply:
Yes we do. Good catch. Will apply the change.

Reply:
Yes, we should adapt `RpcTaskManagerGateway` accordingly.